mantidproject/mantid
# SANS - automatically update default save formats

**Description of work.**
- Automatically updates the selected save formats when toggling between 1D and 2D reduction modes

**Report to:** ISIS SANS / Whole Team

**To test:**
- Download the ISIS training course data, locate `loqdemo`
- Create a empty working folder
- In Mantid change manage user directories output directory to point to this clean directory
- Open ISIS SANS
- Set the user file to either the toml or .txt file in LOQ demo
- Set the batch file to the .csv to the file in that directory
- On the same screen try toggling between 1D and 2D, check that it updates save options each time
- Run process on the first line each time, checking the output dir contains what's expected deleting the files between
- Try toggling to memory / file / both to see that the option is respected

Fixes #31929 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
